include/dpp/stringops.h

```diff
@@ -55,21 +99,36 @@ template <typename T> std::basic_string<T> lowercase(const std::basic_string<T>&
  */
 template <typename T> std::basic_string<T> uppercase(const std::basic_string<T>& s)
 {
-	std::basic_string<T> s2 = s;
+    std::basic_string<T> s2(s);
 	std::transform(s2.begin(), s2.end(), s2.begin(), toupper);
 	return s2;
 }
 
+/**
+ * @brief Convert a string to uppercase using toupper()
+ *
+ * @tparam T type of string
+ * @param s String_view to uppercase
+ * @return std::basic_string<T> uppercased string
+ */
+template <typename T> std::basic_string<T> uppercase(std::basic_string_view<T> s)
+{
+    std::basic_string<T> s2(s);
+    std::transform(s2.begin(), s2.end(), s2.begin(), toupper);
+    return s2;
+}
+
 /**
  * @brief trim from end of string (right)
  * 
  * @param s String to trim
  * @return std::string trimmed string
  */
-inline std::string rtrim(std::string s)
+inline std::string rtrim(std::string_view s)
```

Comment:
I don't agree with this change. You're now having to create a new string which adds more bytes (as you've now got the bytes from `s` + the bytes from `s_cpy`), which now adds 16 more bytes every time you call this function. This should be left as std::string.